src/components/ButtonGroup.tsx

```diff
@@ -0,0 +1,45 @@
+import clsx from 'clsx';
+import { ReactNode } from 'react';
+import { RadioGroup } from '@headlessui/react';
+
+interface ButtonGroupProps {
+  children: ReactNode;
+  value: any;
+  onChange: (selected: any) => void;
+}
```

Comment:
I see below you're using `ReactNode` as the type for `value` in , but isnt that a bit difficult to process in an `onChange` callback?

Regardless, I think some clarity could be added by using a generic type parameter here and in `ButtonGroupOption`

```suggestion
interface ButtonGroupProps<T> {
  children: ReactNode;
  value: T;
  onChange: (selected: T) => void;
}
```